server/src/grpc/meta_event_service/mod.rs

```diff
@@ -362,10 +365,24 @@ async fn do_close_shard(ctx: &HandlerContext, shard_id: ShardId) -> Result<()> {
             msg: "fail to close shards in cluster",
         })?;
 
-    ctx.release_shard_lock(shard_id).await.map_err(|e| {
-        error!("Failed to release shard lock, shard_id:{shard_id}, err:{e}");
-        e
-    })?;
+    if future_ext::retry_async(
```

Comment:
Shall we do retry for the whole `do_close_shard`?

Reply:
Not required, we only need to ensure shard lock is released after cluster.close_shard succeed.